auth/api/iam/openid4vp.go

```diff
@@ -81,8 +87,33 @@ func (r Wrapper) handleAuthorizeRequestFromHolder(ctx context.Context, verifier
 	}
 	// now we have a valid redirectURL, so all future errors will redirect to this URL using the Oauth2ErrorWriter
 
+	// quick check for missing string values that otherwise would blow up
+	for _, param := range []string{oauth.ClientIDParam, oauth.ScopeParam, oauth.StateParam} {
+		if _, ok := params[param].(string); !ok {
+			return nil, withCallbackURI(oauthError(oauth.InvalidRequest, fmt.Sprintf("missing %s parameter", param)), redirectURL)
+		}
+	}
+
+	// additional JAR checks
+	// check if the audience is the verifier
+	audience, ok := params[jwt.AudienceKey].([]string)
+	if !ok || len(audience) == 0 || audience[0] != verifier.String() {
+		return nil, withCallbackURI(oauthError(oauth.InvalidRequest, fmt.Sprintf("invalid audience, verifier = %s, audience = %s", verifier.String(), audience)), redirectURL)
+	}
+	// check if issuer of the JWT equals client_id
+	clientIDClaim := params[oauth.ClientIDParam].(string)
+	issuerClaim, ok := params[jwt.IssuerKey].(string)
+	if !ok || issuerClaim != clientIDClaim {
+		return nil, withCallbackURI(oauthError(oauth.InvalidRequest, "invalid issuer claim"), redirectURL)
```

Comment:
claim/parameter, confusing

Reply:
has been removed since check is done one level higher up